### ISSUE TITLE
Optimization for wordless languages

### ISSUE THREAD

**Issue by aprosail** [vuejs/vitepress#3958]
### Is your feature request related to a problem? Please describe.

When a paragraph is long in markdown, we usually separate them into lines.
But for wordless languages (such as Chinese and Japanese),
an extra line break will cause an unnecessary white space.

Even the current Chinese docs of VitePress puts everything of a long paragraph 
inside a single lone line to prevent such unnecessary white space.
Editing like this makes it really uncomfortable when writing markdown source code.

### Describe the solution you'd like

Preprocess markdown source code to remove line break between a paragraph of wordless language.

### Describe alternatives you've considered

Write a vite plugin to preprocess, it's easy to come up with such idea.
But VitePress had already add the `transform` from markdown to vue in build-in plugins,
so that current API cannot attach the preprocess function in right place.

I've tried to get raw markdown content in the `load` hook,
but such preprocess will parse the markdown source code,
which will also be done in the process from markdown to vue,
that I think it's repeat, and might slow down the performance.
So maybe it's better to add such preprocess inside VitePress source code
rather than make it a plugin.
Or VitePress can provide an extra hook before processing markdown to vue.

### Additional context

1. (This is another idea:) Maybe we can add space between wordless languages and other languages by the way.
2. Chinese and Japanese are not the only wordless languages, maybe we can find more details in the unicode standards.

### Validations

- [X] Follow our [Code of Conduct](https://vuejs.org/about/coc.html)
- [X] Read the [docs](https://vitepress.dev).
- [X] Read the [Contributing Guidelines](https://github.com/vuejs/vitepress/blob/main/.github/contributing.md).
- [X] Check that there isn't already an issue that asks for the same feature to avoid creating a duplicate.

**Comment by aprosail** [comment]
For example, when coding like this,
VitePress already render the markdown source code into vue components,
that the `code` here is not the markdown source code,
but a complex rendered vue component,
which is not easy to modify.

```ts
import {defineConfig} from "vitepress"

export default defineConfig({
  vite: {
    plugins: [
      {
        name: "xxx",
        transform(code, id) {
          if (id.endsWith(".md")) {
            // preprocess code ...
          }
        },
      },
    ],
  },
  // other options ...
})
```

**Comment by aprosail** [comment]
The markdown source code can be accessed in such code, using the `load` hook rather than `transform`.
But I'd like to use some tools, such as `markdown-it`, which is used in VitePress,
and parsing markdown here will be repeat and waste performance.
Calling node file system directly here may also cause some potential risks.

```ts
import {readFileSync} from "node:fs"
import {defineConfig} from "vitepress"

export default defineConfig({
  vite: {
    plugins: [
      {
        name: "vitepress-space",
        load(id) {
          if (id.endsWith(".md")) {
            const code = readFileSync(id).toString()
            // process code ...
          }
        },
      },
    ],
  },
  // other options ...
}
```

**Comment by brc-dd** [comment]
You don't need a vite plugin for this, you need a markdown-it plugin. Also in vite plugins you probably need to add enforce: 'pre' for it to get markdown code. Try something like this:


```ts
import { defineConfig } from 'vitepress'

export default defineConfig({
  markdown: {
    config(md) {
      md.renderer.rules.softbreak = () => ''
    }
  }
})
```

**Comment by aprosail** [comment]
@brc-dd thanks, I'll try it later. Sorry that I'm still new to the VitePress framework.

But by the way, enable word wrap in code editor does not really satisfy my demand.
Break line semantically and structurally makes it easier to read the source code.

**Comment by aprosail** [comment]
@brc-dd thanks for your advices.

I'll close this issue and try to establish a plugin to achieve that.
Just set `softbreak` to empty does not work in all situation,
for example, I usually work with multi-languages within a single file, that I cannot disable all soft breaks.
So I'll try other approaches to make it in details, such as writing a new markdown-it or vite plugin.

**Comment by brc-dd** [comment]
you can get token, etc and do stuff conditionally:

```ts
      md.renderer.rules.softbreak = (tokens, idx, options, env, self) => {
        const token = tokens[idx]
        const content = token.content
        console.log(env.localeIndex)
        return ''
      }
```

**Comment by aprosail** [comment]
@brc-dd Thanks, I've learned and understood the plugin structure. 

And I'm now doing some research on unicode, because there are so many other wordless languages besides the famous Chinese and Japanese, and there are so many special characters in those languages. The `env.localeIndex` you've mentioned before won't work in most situations, it is usually `"root"` rather than exact language code. So I decided to parse it from the unicode range according to corresponding standards. I also wanna to optimize the point of language change in multi-language documentations by the way. I'll mention here once I've finish those tasks.
